### ISSUE TITLE
Local scope translations in component doesn't work

### ISSUE THREAD

**Issue by YarikSWT** [nuxt-modules/i18n#2315]
### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      modules, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0-rc.2
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/nuxt-starter-wm5dcv?file=nuxt.config.ts

### Describe the bug

I was able to define local scope translations for component in Legacy API mode with webpack builder as follows:

```
<template>...{{ $t('message') }}...</template> 
  <script>
     export default defineComponent({
            ...
            i18n: {
               messages: {
                   en: {
                       message: 'Hello, world!'
                   }
               }
            },
            ...
     })
  </script> 
```
  
  But with Vite I can't. It simply doesn't work

**Comment by kazupon** [comment]
Thank you for your reporting!

This is nuxt i18n bug, not webpack issue.
And we need to improve vue-i18n to respect the config `legacy: true`.

BTW, you reproduction code has Composition API mode and Legacy API mode 
Note that vue-i18n does not recommend its use, as stated in the documentation.
https://vue-i18n.intlify.dev/guide/migration/vue3.html

**Comment by libertyswift** [comment]
I have the same problem

**Comment by kazupon** [comment]
I've released v8.0.0-rc.3
if you would like to use legacy API mode, you need to configure `i18n.bundle.compositionOnly: false`.
https://v8.i18n.nuxtjs.org/options/misc#bundle

Happy coding!
